kedro/config/templated_config.py

```diff
@@ -113,24 +114,34 @@ def __init__(
                 dictionary. This dictionary will get merged with the globals dictionary
                 obtained from the globals_pattern. In case of duplicate keys, the
                 ``globals_dict`` keys take precedence.
+            anyconfig_kwargs: Keyword arguments to pass to the the ``anyconfig.load``
+                method. By default we enabled jinja2 via the ``ac_template`` argument.
         """
         super().__init__(
             conf_source=conf_source, env=env, runtime_params=runtime_params
         )
         self.base_env = base_env
         self.default_run_env = default_run_env
+        self._anyconfig_args = (
+            anyconfig_args if anyconfig_args else {"ac_template": True}
+        )
 
         self._config_mapping = (
             _get_config_from_patterns(
                 conf_paths=self.conf_paths,
                 patterns=list(globals_pattern),
-                ac_template=False,
+                anyconfig_args=self._anyconfig_args,
             )
             if globals_pattern
             else {}
         )
+        env_dict = (
+            deepcopy({"KEDRO_ENV": env})
+            if env
+            else deepcopy({"KEDRO_ENV": self.base_env})
+        )
```

Comment:
I don't see the need to `deepcopy` here? Also I don't think this logic is quite right here.  Since we can ignore `base`, I think we should have:
```
self._config_mapping = {**self._config_mapping, **globals_dict, "KEDRO_ENV": env or default_run_env}
```